File: Iteration2/src/Login/LecturerLoginStrategy.java
package Login;

import Domain.Department;
import Domain.Lecturer;
import Domain.RegistrationStatus;

import java.util.Scanner;

public class LecturerLoginStrategy implements ILoginStrategy{
    @Override
    public void login(Department department) {
        loginAsLecturer(department);
    }

    private static void loginAsLecturer(Department department) {
        short var = -1;
        boolean flag = false;
        Lecturer lecturer = null;

        LoginHelper loginHelper = new LoginHelper();

        while (var != 0 && !flag) {


            System.out.println("Enter your staff number");
            System.out.println("0 to cancel");

            String enteredNo;


            try{
                Scanner input2 = new Scanner(System.in);
                enteredNo = input2.next();
            }catch (Exception e){
                System.out.println(e);
                break;
            }
            if(enteredNo.equals("0")){
                return;
            }
            lecturer = department.getLecturerByStaffNo(enteredNo);
            if (lecturer == null) {
                continue;
            }
            flag = true;
        }

        short choice = -1;
        while (choice != 0) {
            System.out.println("0 to exit");
            System.out.println("1 to list teaching courses");
            System.out.println("");
            Scanner input2 = new Scanner(System.in);
            choice = input2.nextShort();
            switch (choice) {
                case 0:
                    break;
                case 1://get active registrations
                    loginHelper.listCoursesForLecturer(department, lecturer);
                    break;
                default://default case
                    break;
            }

        }//While loop end
    }

}

File: Iteration3/Login/LecturerLoginStrategy.py
from Login import ILoginStrategy
from Login.LoginHelper import LoginHelper
from Domain import Department, Lecturer
from Login.ILoginStrategy import ILoginStrategy  # Use relative import
from Login.LoginHelper import LoginHelper  # Assuming LoginHelper is in the same package
import logging
class LecturerLoginStrategy(ILoginStrategy):
    def login(self, department):
        self.login_as_lecturer(department)

    def login_as_lecturer(self, department):
        logger = logging.getLogger()
        var = -1
        flag = False
        lecturer = None
        login_helper = LoginHelper()

        while var != 0 and not flag:
            print("Enter your staff number")
            print("0 to cancel")

            entered_no = input()

            if entered_no == "0":

                return

            lecturer = department.get_lecturer_by_staff_no(entered_no)
            if lecturer is None:
                logger.info(f"Lecturer login unsuccessfull, entered no is {entered_no} is not found")
                continue
            else:
                print("Enter your password:")
                entered_password = input()
                if entered_password != lecturer.get_password():
                    logger.info(f"Lecturer login unsuccessfull, lecturer {entered_no} entered wrong password")
                    print("Wrong password try again")
                    continue
            flag = True
        logger.info(f"Lecturer login successful, lecturer {entered_no} is log inned.")

        choice = -1
        while choice != 0:
            print("0 to exit")
            print("1 to list teaching courses")
            try:
                choice = int(input())
            except ValueError as e:
                print(e)

            if choice == 0:
                break
            elif choice == 1:  # list teaching courses
                login_helper.list_courses_for_lecturer(department, lecturer)
            else:
                # default case
                break
